src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -82,14 +85,16 @@ func (c *IdleTimingConn) TimesOutAt() time.Time {
 // Read implements the method from io.Reader
 func (c *IdleTimingConn) Read(b []byte) (int, error) {
 	totalN := 0
+	readDeadline := time.Unix(0, atomic.LoadInt64(&c.readDeadline))
```

Comment:
@myleshorton I now grab the deadline up-front before entering the loop.  This way, if the client happens to change the deadline while I'm in the middle of looping, it won't have unpredictable effects.